supervisor/datatypes.py

```diff
@@ -344,17 +345,23 @@ def url(value):
 
 # all valid signal numbers
 sig_pattern = re.compile("CTRL|SIG[^_]")
+win_sig_pattern = re.compile("WM_(?!NULL)")
 SIGNUMS = [getattr(signal, k) for k in dir(signal) if sig_pattern.match(k)]
-
+SIGNUMS += [-getattr(win32con, k) for k in dir(win32con) if win_sig_pattern.match(k)]
 
 def signal_number(value):
     try:
         num = int(value)
     except (ValueError, TypeError):
         name = value.strip().upper()
-        if not sig_pattern.match(name):
-            name = 'SIG' + name
-        num = getattr(signal, name, None)
+        if win_sig_pattern.match(name):
```

Comment:
It would be interesting to invert the check because windows have low precedence over the console.
```
if not sig_pattern.match(name):
		name = 'SIG' + name
	num = getattr(signal, name, None)
	
elif win_sig_pattern.match(name):
	num = getattr(win32con, name, None)
	if num is not None:
		num = -num
```

Reply:
This is tricky to do because the pre-existing assumption was that any signal name that doesn't match should be prefixed with `"SIG"` (for example `stopsignal=TERM` previously worked as it was turned into `SIGTERM` but we don't want `WM_CLOSE` to be caught by that logic). I don't see anyway to invert the check without combining the regexes but combined regex makes it impossible to make the Windows events have a negative signal number.

`timeit` suggests the regex check `win_sig_pattern.match(name)` only takes 0.1 microseconds at it's worse performance (0.007 microseconds best performance) so it may be ok to leave as is. Is it OK to leave as it is now, or would you like me to try and find another way to invert it?

Reply:
Okay. I understand now.